src/main/java/com/imatia/campusdual/grupoun_bootcampbackend/service/RoomService.java

```diff
@@ -0,0 +1,95 @@
+import com.imatia.campusdual.grupoun_bootcampbackend.model.dto.dtomapper.RoomMapper;
+import com.imatia.campusdual.grupoun_bootcampbackend.model.entity.Hotel;
+import com.imatia.campusdual.grupoun_bootcampbackend.model.entity.Room;
+import com.imatia.campusdual.grupoun_bootcampbackend.service.exception.HotelDoesNotExistException;
+import com.imatia.campusdual.grupoun_bootcampbackend.service.exception.InvalidAssignedHotelException;
+import com.imatia.campusdual.grupoun_bootcampbackend.service.exception.InvalidRoomNumberException;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service("RoomService")
+@Lazy
+public class RoomService implements IRoomService {
+
+    private final RoomDAO roomDAO;
+    private final HotelService hotelService;
+    private final RoomMapper roomMapper;
+    private final HotelMapper hotelMapper;
+    private final int FIRST_ROOM_NUMBER = 101; // No rooms in the ground floor
+
+    public RoomService(RoomDAO roomDAO, HotelService hotelService, RoomMapper roomMapper, HotelMapper hotelMapper) {
+        this.roomDAO = roomDAO;
+        this.hotelService = hotelService;
+        this.roomMapper = roomMapper;
+        this.hotelMapper = hotelMapper;
+    }
+
+    @Override
+    public int insertRoom(RoomDTO roomDTO) throws InvalidAssignedHotelException, InvalidRoomNumberException {
+        int assignedHotelId = roomDTO.getHotelId();
+
+        if (assignedHotelId == 0 || hotelService.queryAll().stream().noneMatch(hotel -> hotel.getId() == assignedHotelId)) {
+            throw new InvalidAssignedHotelException("The assigned hotel does not exist");
+        }
+
+        // TODO: Hotels have 9 floors at most, hotel creation validation needed (?)
+        int floorNumber = roomDTO.getFloorNumber();
+        HotelDTO assignedHotelDTO = new HotelDTO();
+        assignedHotelDTO.setId(assignedHotelId);
+        assignedHotelDTO = hotelService.queryHotel(assignedHotelDTO);
+        if (
+                roomDTO.getRoomNumber() < FIRST_ROOM_NUMBER ||
+                        floorNumber > assignedHotelDTO.getNumberOfFloors() ||
+                        queryAll()
+                                .stream()
+                                .anyMatch(
+                                        room -> room.getRoomNumber() == roomDTO.getRoomNumber() &&
+                                                room.getHotelId() == roomDTO.getHotelId()
+                                )
+        ) {
+            throw new InvalidRoomNumberException("Cannot create room with this number");
+        }
+
+        Hotel assignedHotel = hotelMapper.toEntity(hotelService.queryHotel(assignedHotelDTO));
+        Room room = roomMapper.toEntity(roomDTO);
+        room.setHotel(assignedHotel);
+        room = roomDAO.saveAndFlush(room);
+        assignedHotel.getRooms().add(room);
+        try {
+            hotelService.updateHotel(hotelMapper.toDTO(assignedHotel));
```

Comment:
Por que se actualiza o hotel cando se inserta unha nova habitación?

Reply:
A habitación está relacionada con `Hotel` cunha relación one-to-many e especificouse que para esa relación as operacións de persistencia se propaguen en cascada (`CascadeType.ALL`). Ao gardar un hotel desencadéase o gardado das súas habitacións asociadas.

Reply:
Fixémolo así porque entendimos que se modifica o hotel porue se lle engade unha nova habitación que non pode existir sen o hotel asignado, pero tamén se pode cambiar a un gardado independente sen maior problema.

Reply:
esta ben visto pero esa comprobación tamen a fas un pouco antes e sería redundante

Reply:
Si, tal como o fixemos sobraríanos a liña 65 na que se garda a habitación por separado ou o bloque try-catch no que se garda o hotel e a habitación en cadea. Usamos a solución que nos aconselles ti.

Reply:
eu diriavos que quitasedes a actualización do hotel se non se cambiou ningun dato

Reply:
Modificado.